lib/rucio/api/rule.py

```diff
@@ -318,5 +318,4 @@ def move_replication_rule(rule_id, rse_expression, activity, source_replica_expr
 
     return rule.move_rule(rule_id=rule_id,
                           rse_expression=rse_expression,
-                          activity=activity,
-                          source_replica_expression=source_replica_expression)
+                          override={'activity': activity, 'source_replica_expression': source_replica_expression})
```

Comment:
I have the impression that this won’t work properly. In the CLI, `--activity` and `--source-replica-expression` are optional. I guess this means, when not specified, that the client will send `None`? If so, won’t this use of `override` actually replace the parameters of the original rule?

I think what needs to be done here is 1) populate `override` only when necessary and 2) have some special value which will actually translate to `None`. I propose doing in the client something similar to `update_rule`.

https://github.com/rucio/rucio/blob/d59c237f533f40116026dc9f347f4fc1297f1ff0/bin/rucio#L1419-L1420

Reply:
I think we don't have to replace the paraemters at all when no `override` value is given, since the default values come from the original rule.

Do you confirm that @dchristidis?

Reply:
Forgive me, I do not understand your question.

In my mind, two things should be possible from the CLI:

```
$ rucio move-rule RULE_ID RSE_EXPRESSION
$ rucio move-rule --source-replica-expression None RULE_ID RSE_EXPRESSION
```

I believe the latest changes fixed the first command, but not the second?

Reply:
Ah now i understand what you mean. You're right, the second one would not be possible right now. We have to change the command line argument itself since there is not difference between not specifying it and the value `None` right now. I'll have a look into that.

Reply:
Okay I though about it and would rather introduce new command line arguments to disable/remove the `source-replica-expression` and `activity`. (Something like `--without-source-replica-expression` or `--no-source-replica-expression` or `--set-source-replica-expression-none`)

If we rather want to overload the current one, we have to introduce a new `"None"` state and have to properly document it. The user has to know that `None` as value would do something different from not specifying it, and I think this could create some problems.

Reply:
Thoughts about that?

Reply:
Your proposal is sensible. Personally, I have a preference for having a special value like `None` or an empty argument because I feel like it’s easier to remember, but I understand how it can be problematic.

Whatever we decide, I think it’s important to apply it consistently. Given that `rucio update-rule --lifetime None RULE_ID` has existed for some time, I think it’s worth discussing it during a DDM DevOps or Rucio meeting.

Reply:
Yes I'm on your side, it should be consistent! I'll implement it with a special `None` value.

Yeah we could ask that in a Rucio meeting, it should not be that important though. As you said consistency is the only important thing here.